### PR TITLE
docker_swarm: more integration tests

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Adds integration tests for most docker_swarm options. Missing are options needed for joining/leaving swarms (that's hard to test if you just have one node) and options handling with CA keys/certs.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
docker_swarm

### CODE REVIEW

**Comment by felixfontein** [comment]
Four CI errors:
 - two times "context deadline exceeded" during docker swarm removal (https://app.shippable.com/github/ansible/ansible/runs/110454/57/tests, https://app.shippable.com/github/ansible/ansible/runs/110454/58/tests);
 - two times "read timeout" during swarm startup/shutdown (https://app.shippable.com/github/ansible/ansible/runs/110454/60/tests, https://app.shippable.com/github/ansible/ansible/runs/110454/65/tests).
I'm wondering whether all these tests somehow degenerate the docker daemon. Maybe some random sleeping inbetween will help? Or not running all of them by default?

**Comment by felixfontein** [comment]
I moved the fixes contained in this PR into #53039 and #53038 so they can be merged without fixing all the tests first.

**Comment by felixfontein** [comment]
ready_for_review

**Review by gundalow — COMMENTED** [review]
From the discussion on IRC, we haven't seen any OS specific Ansible+Docker bugs, so `skip/docker` should be OK.

File: test/integration/targets/docker_swarm/aliases [hunk 1]
```diff
@@ -3,3 +3,5 @@ skip/osx
 skip/freebsd
 destructive
 skip/rhel8.0
+skip/docker
```
Comment (mattclay) [inline test/integration/targets/docker_swarm/aliases]: Append a comment to the end of this line to explain why the tests are skipped.

Comment (felixfontein) [reply test/integration/targets/docker_swarm/aliases]: Done. I didn't knew one can put comments into the `aliases` file...

**Comment by felixfontein** [comment]
shipit

**Comment by felixfontein** [comment]
@gundalow @mattclay thanks for reviewing!

**Pull request merged**